### PR TITLE
upgrade: deno to v0.8.0

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: archive/tar.ts [hunk 1]
```diff
@@ -457,15 +457,15 @@ export class Untar {
       (key): void => {
         const arr = trim(header[key]);
         if (arr.byteLength > 0) {
-          meta[key as keyof UntarOptions] = parseInt(decoder.decode(arr), 8);
```
Comment (piscisaureus) [inline archive/tar.ts]: The error you're trying to solve here is a bit misleading.
The real problem is that `UntarOptions` has a property (`fileName`) which has type `string`.
So typescript reasons that `key` *might* be a `fileName`, and refuses to assign an integer to a string-typed property

Comment (ry) [reply archive/tar.ts]: I'm stuck trying to solve this. Can you give me a hint?

Comment (kitsonk) [reply archive/tar.ts]: It is bit verbose, but it works... Once TypeScript "knows" you are only using certain keys, it allows the assignment.  Currently, IIRC, there is no way for TypeScript to interpret it as a literal array for type checking, so therefore the explicit cast:

```diff
diff --git a/archive/tar.ts b/archive/tar.ts
index c72abc6..5698ae6 100644
--- a/archive/tar.ts
+++ b/archive/tar.ts
@@ -453,19 +453,24 @@ export class Untar {
     if (fileNamePrefix.byteLength > 0) {
       meta.fileName = decoder.decode(fileNamePrefix) + "/" + meta.fileName;
     }
-    ["fileMode", "mtime", "uid", "gid"].forEach(
+    (["fileMode", "mtime", "uid", "gid"] as [
+      "fileMode",
+      "mtime",
+      "uid",
+      "gid"
+    ]).forEach(
       (key): void => {
         const arr = trim(header[key]);
         if (arr.byteLength > 0) {
-          meta[key as keyof UntarOptions] = parseInt(decoder.decode(arr), 8);
+          meta[key] = parseInt(decoder.decode(arr), 8);
         }
       }
     );
-    ["owner", "group"].forEach(
+    (["owner", "group"] as ["owner", "group"]).forEach(
       (key): void => {
         const arr = trim(header[key]);
         if (arr.byteLength > 0) {
-          meta[key as keyof UntarOptions] = decoder.decode(arr);
+          meta[key] = decoder.decode(arr);
         }
       }
     );
```

Comment (ry) [reply archive/tar.ts]: Thanks!

**Pull request merged**